cmd/cstorvolumeclaim/controller.go

```diff
@@ -385,3 +376,192 @@ func BaseLabels(cvc *apis.CStorVolumeClaim) map[string]string {
+			newConditions = append(newConditions, condition)
+			continue
+		}
+
+		if newCondition, ok := resizeConditionMap[condition.Type]; ok {
+			newConditions = append(newConditions, newCondition.condition)
+			newCondition.processed = true
+		}
+	}
+	// append all unprocessed conditions
+	for _, newCondition := range resizeConditionMap {
+		if !newCondition.processed {
+			newConditions = append(newConditions, newCondition.condition)
+		}
+	}
+	return newConditions
+}
+
+func (c *CVCController) markCVCResizeFinished(cvc *apis.CStorVolumeClaim) error {
+	newCVC := cvc.DeepCopy()
+	newCVC.Status.Capacity = cvc.Spec.Capacity
+
+	newCVC.Status.Conditions = MergeResizeConditionsOfCVC(cvc.Status.Conditions, []apis.CStorVolumeClaimCondition{})
+	_, err := c.PatchCVCStatus(cvc, newCVC)
+	if err != nil {
+		glog.Errorf("Mark CVC %q as resize finished failed: %v", cvc.Name, err)
+		return err
+	}
+
+	glog.V(4).Infof("Resize CVC %q finished", cvc.Name)
+	c.recorder.Eventf(cvc, corev1.EventTypeNormal, string(apis.CStorVolumeClaimResizeSuccess), "Resize volume succeeded")
+
+	return nil
+}
+
+// PatchCVCStatus updates CVC status using patch api
+func (c *CVCController) PatchCVCStatus(oldCVC,
+	newCVC *apis.CStorVolumeClaim,
+) (*apis.CStorVolumeClaim, error) {
+	patchBytes, err := getPatchData(oldCVC, newCVC)
+	if err != nil {
+		return nil, fmt.Errorf("can't patch status of CVC %s as generate path data failed: %v", oldCVC.Name, err)
+	}
+	updatedClaim, updateErr := c.clientset.OpenebsV1alpha1().CStorVolumeClaims(oldCVC.Namespace).
+		Patch(oldCVC.Name, types.MergePatchType, patchBytes)
+
+	if updateErr != nil {
+		return nil, fmt.Errorf("can't patch status of CVC %s with %v", oldCVC.Name, updateErr)
+	}
+	return updatedClaim, nil
+}
+
+func getPatchData(oldObj, newObj interface{}) ([]byte, error) {
+	oldData, err := json.Marshal(oldObj)
+	if err != nil {
+		return nil, fmt.Errorf("marshal old object failed: %v", err)
+	}
+	newData, err := json.Marshal(newObj)
+	if err != nil {
+		return nil, fmt.Errorf("mashal new object failed: %v", err)
+	}
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, oldObj)
+	if err != nil {
+		return nil, fmt.Errorf("CreateTwoWayMergePatch failed: %v", err)
+	}
+	return patchBytes, nil
+}
+
+// resizeVolume resize the volume to desired size, and update CV's capacity
+//func (c *CVCController) resizeVolume(
+//	cvc *apis.CStorVolumeClaim,
+//	cv *apis.CStorVolume) error {
+//
+//	newSize := cvc.Spec.Capacity[corev1.ResourceStorage]
+//	err := c.UpdateCVCapacity(cv, newSize)
+//	if err != nil {
+//		glog.Errorf("failed to update capacity of CV %q to %s : %v", cv.Name, newSize.String(), err)
+//		return err
+//	}
+//	glog.V(4).Infof("successfully Updated capacity of CV %q to %s", cv.Name, newSize.String())
+//	return err
+//}
+
+// resizeVolume resize the volume to desired size, and update CV's capacity
+func (c *CVCController) resizeVolume(cv *apis.CStorVolume, newCapacity resource.Quantity) error {
```

Comment:
how about changing fn name to `resizeCV`?

Reply:
done